receiver/dockerstatsreceiver/receiver_test.go

```diff
@@ -120,6 +121,48 @@ var (
 	}
 )
 
+func TestDockerHostOption(t *testing.T) {
+	tests := []struct {
+		name        string
+		endpoint    string
+		expectEnv   bool
+		description string
+	}{
+		{
+			name:        "Empty endpoint, DOCKER_HOST set",
+			endpoint:    "",
+			expectEnv:   true,
+			description: "Should append WithHostFromEnv() when Endpoint is empty.",
+		},
+		{
+			name:        "Empty endpoint and empty env",
+			endpoint:    "",
+			expectEnv:   true,
+			description: "Should append WithHostFromEnv() even if DOCKER_HOST is empty.",
+		},
+		{
+			name:        "Config endpoint set, DOCKER_HOST ignored",
+			endpoint:    "tcp://config:1234",
+			expectEnv:   false,
+			description: "Should not append WithHostFromEnv() when Endpoint is set.",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			var opts []client.Opt
```

Comment:
This test is not validating behavior of the receiver anymore, just the behavior of code copied from the receiver.

If you refactor the receiver to have a function that calculates the options, that could then be tested here.

```
func (r *metricsReceiver) clientOptions() []client.Opt {
  var opts []client.Opt
  if r.config.Config.Endpoint == "" {
    opts = append(opts, client.WithHostFromEnv())
  }
  return opts
}
```